src/detail/filter.hpp

```diff
@@ -131,22 +119,13 @@ class BOOST_HTTP_PROTO_DECL
         return on_process(out, in, more);
     }
 
-    results
-    process_impl(
-        buffers::mutable_buffer_span const& out,
-        buffers::const_buffer_span const& in,
-        bool more)
-    {
-        return on_process(out, in, more);
-    }
-
     template<
-        class MutableBuffers,
-        class ConstBuffers>
+        class MutableBufferSequence,
+        class ConstBufferSequence>
     results
     process_impl(
-        MutableBuffers const& out,
-        ConstBuffers const& in,
+        MutableBufferSequence const& out,
+        ConstBufferSequence const& in,
```

Comment:
Do Asio-style buffer sequences work when passed to this?

Reply:
Yes, except here we are using `buffers::begin` and `buffers::end`. By the way, the filter is an implementation detail, so a user-passed buffer sequence would never encounter it directly.